### PR TITLE
PhysFormer Support

### PR DESCRIPTION
**Pull request by yahskapar**
This PR revolves around PhysFormer model support, including train configs, infer configs, and pre-trained models that were used for results in a pending update to the pre-print associated with the rPPG-Toolbox. Please see the individual commits for more details.

### CODE REVIEW

File: neural_methods/model/PhysFormer.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+        )
+        self.proj_v = nn.Sequential(
+            nn.Conv3d(dim, dim, 1, stride=1, padding=0, groups=1, bias=False),  
+            #nn.BatchNorm3d(dim),
+            #nn.ELU(),
+        )
+        
+        #self.proj_q = nn.Linear(dim, dim)
+        #self.proj_k = nn.Linear(dim, dim)
+        #self.proj_v = nn.Linear(dim, dim)
```
Comment (xliucs) [inline neural_methods/model/PhysFormer.py:92]: Remove them.

File: neural_methods/model/PhysFormer.py [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+from torch.nn import functional as F
+import math
+
+def as_tuple(x):
+    return x if isinstance(x, tuple) else (x, x)
+
+'''
+Temporal Center-difference based Convolutional layer (3D version)
+theta: control the percentage of original convolution and centeral-difference convolution
+'''
```
Comment (xliucs) [inline neural_methods/model/PhysFormer.py:23]: Why there is a multi-line string after a method def?

File: neural_methods/model/PhysFormer.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+        )
+        
+        self.fc2 = nn.Sequential(
+            nn.Conv3d(ff_dim, dim, 1, stride=1, padding=0, bias=False),  
+            nn.BatchNorm3d(dim),
+        )
+
+    def forward(self, x):    # [B, 4*4*40, 128]
+        [B, P, C]=x.shape
+        #x = x.transpose(1, 2).view(B, C, 40, 4, 4)      # [B, dim, 40, 4, 4]
```
Comment (xliucs) [inline neural_methods/model/PhysFormer.py:152]: remove it.

File: neural_methods/model/PhysFormer.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+        x = x.transpose(1, 2).view(B, C, P//16, 4, 4)      # [B, dim, 40, 4, 4]
+        x = self.fc1(x)		              # x [B, ff_dim, 40, 4, 4]
+        x = self.STConv(x)		          # x [B, ff_dim, 40, 4, 4]
+        x = self.fc2(x)		              # x [B, dim, 40, 4, 4]
+        x = x.flatten(2).transpose(1, 2)  # [B, 4*4*40, dim]
+        
+        return x
+        
+        # (B, S, D) -> (B, S, D_ff) -> (B, S, D)
+        #return self.fc2(F.gelu(self.fc1(x)))
```
Comment (xliucs) [inline neural_methods/model/PhysFormer.py:162]: remove .

File: neural_methods/model/PhysFormer.py [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+        x = self.patch_embedding(x)  # [B, 64, 40, 4, 4]
+        x = x.flatten(2).transpose(1, 2)  # [B, 40*4*4, 64]
+        
+        
+        Trans_features, Score1 =  self.transformer1(x, gra_sharp)  # [B, 4*4*40, 64]
+        Trans_features2, Score2 =  self.transformer2(Trans_features, gra_sharp)  # [B, 4*4*40, 64]
+        Trans_features3, Score3 =  self.transformer3(Trans_features2, gra_sharp)  # [B, 4*4*40, 64]
+        
+        
+        # Trans_features3 = self.normLast(Trans_features3)
```
Comment (xliucs) [inline neural_methods/model/PhysFormer.py:313]: remove.

**Review by xliucs — COMMENTED** [review]
Can you also update README file to reflect this new model support? 

Looks great to me in general. I left some minor comments.

**Comment by xliucs** [comment]
Also, can you also include pre-trained models if the file size is small?

File: neural_methods/model/PhysFormer.py (hunk 1 above)
Comment (yahskapar) [reply neural_methods/model/PhysFormer.py:92]: Addressed in a commit that will be pushed shortly.

File: neural_methods/model/PhysFormer.py (hunk 3 above)
Comment (yahskapar) [reply neural_methods/model/PhysFormer.py:152]: Addressed in a commit that will be pushed shortly.

File: neural_methods/model/PhysFormer.py (hunk 4 above)
Comment (yahskapar) [reply neural_methods/model/PhysFormer.py:162]: Addressed in a commit that will be pushed shortly.

**Comment by yahskapar** [comment]
> Also, can you also include pre-trained models if the file size is small?

These are already included in this PR in [this commit](https://github.com/ubicomplab/rPPG-Toolbox/pull/203/commits/896f15fcef3d6e9d1ab0dc0187b19881d12032d8). Seems like the file size addition to the overall repo should be fine for now.

**Comment by yahskapar** [comment]
Updated PR as requested @xliucs.

**Review by xliucs — APPROVED** [review]
LGTM

**Pull request merged**